### PR TITLE
Refactor FXIOS-9820 Enable Swiftlint rules for SwiftUI accessibility

### PR DESCRIPTION
**Pull request by thatswinnie**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9820)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21568)

## :bulb: Description
Enables the Swiftlint rules to check for button and image accessibility and fixes the warnings resulting from that.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/WidgetKit/ImageButtonWithLabel.swift [hunk 1]
```diff
@@ -90,19 +90,19 @@ struct ImageButtonWithLabel: View {
                         }
                         Spacer()
                         if link == .search && isSmall {
-                            Image("searchLarge")
+                            Image(decorative: "searchLarge")
```
Comment (razvanlitianu) [inline firefox-ios/WidgetKit/ImageButtonWithLabel.swift:93]: Just found out about this Image [init](https://developer.apple.com/documentation/swiftui/image/init(decorative:bundle:)). Does it mean the VoiceOver will skip it?

Comment (thatswinnie) [reply firefox-ios/WidgetKit/ImageButtonWithLabel.swift:93]: @razvanlitianu Yes, decorative images will be skipped in VoiceOver. Please let me know if any of the images in this PR should not be skipped and should have an a11y label instead.

File: firefox-ios/WidgetKit/OpenTabs/OpenTabsWidget.swift [hunk 2]
```diff
@@ -81,7 +81,7 @@ struct OpenTabsView: View {
                     Text(String.NoOpenTabsLabel)
                     HStack {
                         Spacer()
-                        Image("externalLinkSmall")
+                        Image(decorative: "externalLinkSmall")
```
Comment (cyndichin) [inline firefox-ios/WidgetKit/OpenTabs/OpenTabsWidget.swift:84]: nit: not part of this PR since pre-existing, but should these hardcoded image strings live elsewhere? also seems like its duplicate, so updating this can have a single place where it's updated.

**Review by cyndichin — APPROVED** [review]
yaaay, thanks for adding this enhancement @thatswinnie !! 🔥 🔥 🔥

**Review by razvanlitianu — APPROVED** [review]

File: firefox-ios/WidgetKit/OpenTabs/OpenTabsWidget.swift (hunk 2 above)
Comment (thatswinnie) [reply firefox-ios/WidgetKit/OpenTabs/OpenTabsWidget.swift:84]: Good point. I think we had them hardcoded in the past because the widgets and extensions didn't include `Common` and also they have their own asset catalogs (I think because of size restrictions for extensions). I'll create a task to use `StandardImageIdentifiers` here as well.

**Review by adudenamedruby — APPROVED** [review]

**Review by clarmso — APPROVED** [review]
